Azure/sonic-buildimage
# [WIP] [mclagsyncd] added peer link state

#### Why I did it

In case peer-link is down mclag feature is not properly working, but `mclagdctl` doesn't show any error, all statuses are `OK`. The system will be more traceable when `mclagdctl` will show peer-link state. 

```

# MCLAG is working properly, some traffic will be forwarded/received using peer-link.

$ mclagdctl dump state
The MCLAG's keepalive is: OK
MCLAG info sync is: completed
Domain id: 1
Local Ip: 192.168.3.1
Peer Ip: 192.168.3.2
Peer Link Interface: PortChannel0002
Keepalive time: 1
sesssion Timeout : 15
Peer Link Mac: 52:51:00:12:34:56
Role: Active
MCLAG Interface: PortChannel0001
Loglevel: NOTICE

$ config interface shut PortChannel0002

# MCLAG is not working properly, peer-link is down, some traffic will be lost, but it is not traceable.

$ mclagdctl dump state
The MCLAG's keepalive is: OK
MCLAG info sync is: completed
Domain id: 1
Local Ip: 192.168.3.1
Peer Ip: 192.168.3.2
Peer Link Interface: PortChannel0002
Keepalive time: 1
sesssion Timeout : 15
Peer Link Mac: 52:51:00:12:34:56
Role: Active
MCLAG Interface: PortChannel0001
Loglevel: NOTICE

```

#### How I did it

 - add `peer_link_state` to struct `mclagd_state`
 - parse `peer_link_state` in function `iccp_mclag_config_dump`
 - show `peer_link_state` in function `mclagdctl_parse_dump_state`

#### How to verify it

- setup MCLAG topology
- run `mclagdctl dump state` ,  keepalive must be `OK` and `Peer Link state: Up`, expected output:
```
$ mclagdctl dump state
The MCLAG's keepalive is: OK
MCLAG info sync is: completed
Domain id: 1
Local Ip: 192.168.3.1
Peer Ip: 192.168.3.2
Peer Link Interface: PortChannel0002
Peer Link state: Up
Keepalive time: 1
sesssion Timeout : 15
Peer Link Mac: 52:51:00:12:34:56
Role: Active
MCLAG Interface: PortChannel0001
Loglevel: NOTICE
```

- shutdown peer link port `config interface shut PortChannelForPeerLink`
- run `mclagdctl dump state` ,  keepalive must be `OK` and `Peer Link state: Down`, expected output:
```
$ mclagdctl dump state
The MCLAG's keepalive is: OK
MCLAG info sync is: completed
Domain id: 1
Local Ip: 192.168.3.1
Peer Ip: 192.168.3.2
Peer Link Interface: PortChannel0002
Peer Link state: Down
Keepalive time: 1
sesssion Timeout : 15
Peer Link Mac: 52:51:00:12:34:56
Role: Active
MCLAG Interface: PortChannel0001
Loglevel: NOTICE
```


#### Which release branch to backport (provide reason below if selected)


#### Description for the changelog

`mclagdctl dump state` will additionally show peer-link port state.
